File: 2107-find-unique-binary-string/find-unique-binary-string.cpp
class Solution {
public:
vector<string>ans;
int n;
 unordered_map<string,int>mp;
 void solve(int i,string s)
 {  if(ans.size()>0)
 {
    return ;
 }
    if(i==n)
    {
        if(mp[s]<1)
        {
            ans.push_back(s);
            return ;
        }
        else
        {
            return ;
        }
    }
    s.push_back('0');
    solve(i+1,s);
    s.pop_back();
    s.push_back('1');
    solve(i+1,s);
    s.pop_back();
    return ;
 }
    string findDifferentBinaryString(vector<string>& nums) {
      n=nums.size();  
     
      for(int i=0;i<nums.size();i++)
      {
        mp[nums[i]]++;
      }
      solve(0,"");
      return ans[0];
    }
};

File: 2107-find-unique-binary-string/find-unique-binary-string.java
class Solution {
    String ans = "";

    void solve(int n, HashMap<String, Integer> mp, StringBuilder temp) {

        if (temp.length() == n) {
            String s = temp.toString();
            if (!mp.containsKey(s)) {
                ans = s;
            }
            return;
        }

        if (!ans.equals("")) {
            return;
        }

        temp.append('0');
        solve(n, mp, temp);
        temp.deleteCharAt(temp.length() - 1);

        temp.append('1');
        solve(n, mp, temp);
        temp.deleteCharAt(temp.length() - 1);
    }

    public String findDifferentBinaryString(String[] nums) {

        HashMap<String, Integer> mp = new HashMap<>();
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            mp.put(nums[i], 1);
        }

        StringBuilder temp = new StringBuilder();

        solve(n, mp, temp);

        return ans;
    }
}